src/contributors/__tests__/prompt.js

```diff
@@ -0,0 +1,68 @@
+import prompt from '../prompt'
+
+function fixtures() {
+  const options = {
+    contributors: [
+      {
+        login: 'jfmengels',
+        name: 'Jeroen Engels',
+        avatar_url: 'https://avatars.githubusercontent.com/u/3869412?v=3',
+        profile: 'https://github.com/jfmengels',
+        contributions: [Array],
+      },
+      {
+        login: 'kentcdodds',
+        name: 'Kent C. Dodds',
+        avatar_url: 'https://avatars.githubusercontent.com/u/1500684?v=3',
+        profile: 'http://kentcdodds.com/',
+        contributions: [Array],
+      },
+      {
+        login: 'jccguimaraes',
+        name: 'João Guimarães',
+        avatar_url: 'https://avatars.githubusercontent.com/u/14871650?v=3',
+        profile: 'https://github.com/jccguimaraes',
+        contributions: [Array],
+      },
+    ],
+  }
+  return options
+}
+
+test(`should throw error if all contribution types are invalid`, () => {
+  const options = fixtures()
+  const username = 'userName'
+  const contributions = 'invalidContributionType1,invalidContributionType2'
+  expect(() => prompt(options, username, contributions)).toThrow(
+    'Invalid contribution type/s entered',
+  )
+})
+
+test(`should not throw error if atleast one of the contribution types is valid`, () => {
+  const options = fixtures()
+  const username = 'userName'
+  const contributions = 'wrongContributionType,code'
+  return prompt(options, username, contributions).then(answers => {
+    expect(answers).toEqual({username: 'userName', contributions: ['code']})
+  })
+})
+
+test(`should filter valid contribution types from user inserted types`, () => {
+  const options = fixtures()
+  const username = 'userName'
+  const contributions =
+    'invalidContributionType1,code,invalidContributionType2,bug'
+  return prompt(options, username, contributions).then(answers => {
+    expect(answers.contributions.length).toBe(2)
+    expect(answers.contributions).toEqual(['code', 'bug'])
+  })
+})
+
+test(`should throw error if no contribution types were included in add command`, () => {
+  const options = fixtures()
+  const username = 'userName'
+  const contributions = ''
+  expect(() => prompt(options, username, contributions)).toThrow(
+    `No contribution type found in the input. Did you forget to include them in the add command?`,
+  )
```

Comment:
If no contribution types are found in the add command, the user should be prompted for them.

I should be able to just run the cli without any argument, be prompted for the operation I want to do (add/generate), then be prompted for the rest of the arguments.

Currently your patch breaks that:

<img width="658" alt="screen shot 2017-12-25 at 12 20 13 pm" src="https://user-images.githubusercontent.com/304450/34338927-23602e3a-e96e-11e7-9388-389bb111256a.png">

Reply:
ok let us handle this in another pr then